app/controllers/application_controller.rb

```diff
@@ -3,12 +3,26 @@ class ApplicationController < ActionController::API
   private
 
   set_current_tenant_through_filter
+  before_action :set_current_headers
```

Comment:
@syncrou Can we have them setup/teardown with a functions that do both the calls

 def set_current
    set_current_headers
    set_current_tenant
end

def unset_current
   ManageIQ::API::Common::Request.current = nil
   ActsAsTenant.current_tenant = nil
end
 
before_action :set_current
after_action :unset_current